controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/CreateReaderGroupTask.java

```diff
@@ -57,48 +65,78 @@ public CreateReaderGroupTask(final StreamMetadataTasks streamMetaTasks,
-                                                  null, executor);
-                                      }, executor);
-                                   }
-                               return CompletableFuture.completedFuture(null);
-                               }).thenCompose(v ->
-                                  streamMetadataTasks.createRGStream(scope, NameUtils.getStreamForReaderGroup(readerGroup),
-                                           StreamConfiguration.builder().scalingPolicy(ScalingPolicy.fixed(1)).build(),
-                                          System.currentTimeMillis(), 10)
-                                 .thenCompose(status -> {
-                                     if (status.equals(Controller.CreateStreamStatus.Status.STREAM_EXISTS)
-                                     || status.equals(Controller.CreateStreamStatus.Status.SUCCESS)) {
-                                         return Futures.toVoid(streamMetadataStore.updateReaderGroupVersionedState(scope, readerGroup,
-                                                 ReaderGroupState.ACTIVE, state, context, executor));
-                                     }
-                               return Futures.failedFuture(new IllegalStateException(String.format("Error creating StateSynchronizer Stream for Reader Group %s: %s",
-                                             readerGroup, status.toString())));
-                           })).exceptionally(ex -> {
-                                    log.debug(ex.getMessage());
-                                    Throwable cause = Exceptions.unwrap(ex);
-                                    throw new CompletionException(cause);
-                           });
+        return RetryHelper.withRetriesAsync(() -> streamMetadataStore.getReaderGroupId(scope, readerGroup, context, executor)
+                .thenCompose(rgId -> {
+                    if (!rgId.equals(readerGroupId)) {
+                        log.warn("Skipping processing of CreateReaderGroupEvent with invalid UUID.");
+                        return CompletableFuture.completedFuture(null);
                     }
-                    return CompletableFuture.completedFuture(null);
+                    ReaderGroupConfig config = getConfigFromEvent(request);
+                    return Futures.exceptionallyExpecting(streamMetadataStore.getReaderGroupState(scope, readerGroup, true, null, executor),
+                            e -> Exceptions.unwrap(e) instanceof StoreException.DataNotFoundException, ReaderGroupState.UNKNOWN)
+                            .thenCompose(state -> {
+                                if (state.equals(ReaderGroupState.UNKNOWN) || state.equals(ReaderGroupState.CREATING)) {
+                                    return streamMetadataStore.createReaderGroup(scope, readerGroup, config, System.currentTimeMillis(), context, executor)
```

Comment:
this code is very similar to the one in local controller.. lets consolidate it at a single place please.

Reply:
done